lib/datadog/core/remote/configuration/repository.rb

```diff
@@ -91,10 +91,17 @@ def set(**options)
           end
 
           module Operation
-            class Delete
+            class Base
+              def apply(repository)
+                raise NoMethodError
+              end
+            end
```

Comment:
I presume this is for the sake of typing?

A better approach would be to [create a union type alias](https://github.com/ruby/rbs/blob/0dd11d38c11f55ead0a0d78bb9b19d2eaf3a2104/docs/syntax.md#type-alias-declaration) in `repository.rbs`:

```
module Operation
    ...
end

type operation = Operation::Insert | Operation::Update | Operation::Delete | Operation::Set
```

and then:

```
attr_reader operations: Array[operation]
```

Reply:
Or maybe [as an interface](https://github.com/ruby/rbs/blob/0dd11d38c11f55ead0a0d78bb9b19d2eaf3a2104/docs/syntax.md#interface-declaration), which would enforce `apply` being defined without repetition:

```
interface _Operation
  def apply: (Repository repository) -> void
end

module Operation
  class Insert
    include _Operation
  end
end
```

then:

```
attr_reader operations: Array[_Operation]
```